visa/BitSet.cpp

```diff
@@ -109,27 +112,57 @@ void BitSet::invert(void)
 }
 
 template <typename T>
-void vector_and(T *__restrict__ p1, const T *const p2, unsigned n)
+void vector_and(T *__restrict__ p1, const T *const p2, int n)
 {
-    for (unsigned i = 0; i < n; ++i)
+    int i = 0;
+#ifdef __SSE2__
```

Comment:
is there some other way to make VC/gcc vectorize this loop? i'm not sure introducing these intrinsics manually in code base is a good idea. also, do we know how much improvement this particular change gets us?

Reply:
For gcc, we can tell it to do vectorize via `__attribute__((optimize("tree-vectorize")))`. According to godbolt, MSVC and clang automatically use SSE2 in this loop when compiled with O2.